pkg/client/interfaces.go

```diff
@@ -82,27 +82,72 @@ type Writer interface {
 // StatusClient knows how to create a client which can update status subresource
 // for kubernetes objects.
 type StatusClient interface {
-	Status() StatusWriter
+	Status() SubResourceWriter
 }
 
-// StatusWriter knows how to update status subresource of a Kubernetes object.
-type StatusWriter interface {
+// SubResourceClient knows how to create a client which can update subresource
+// for kubernetes objects.
+type SubResourceClient interface {
```

Comment:
Should we add a TODO comment for SubResourceReader and those non-CRUD subresources?

Reply:
Not sure, I think by then we would rename the `SubResourceWriter`